### ISSUE TITLE
Output of Tensorflow YOLO V5 

### ISSUE THREAD

**Issue by phsilvafct** [ultralytics/yolov5#12368]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hello there,
I am trying to convert yolov5 to tensorflow but the output is not the expected. First of all I ran:

`!python detect.py --weights /content/drive/MyDrive/yolov5/yolov5s.pt --conf 0.5 --source /content/dog_640.jpg`

The predictions from this model: 
![dog_640 (1)](https://github.com/ultralytics/yolov5/assets/70081290/8ee344f5-9852-4181-8f64-a41ef8302985)

With the model, I tried to attain the results of the detection in Google Colab. To do so I exported the Tensorflow model with:

`!python export.py --weights /content/drive/MyDrive/yolov5/yolov5s.pt --include tfjs `

In Colab, I converted the image to an array with the dimensions to match the input [1, 640, 640, 3]:

 ```
input_image = keras.preprocessing.image.load_img('/content/dog_640.jpg', target_size=(640, 640))
 input_image = keras.preprocessing.image.img_to_array(input_image)
 input_image = np.expand_dims(input_image, axis=0)

saved_model = tf.saved_model.load('/content/drive/MyDrive/yolov5/yolov5s_saved_model')

predictions = saved_model(input_image) 
```

This shape of these predictions is a list with [[1,100, 4], [1,100], [1,100], [100]]. My problem is interpreting these values, for the first element of the list it is clear it is related to the bounding box:

  [ 0.66175324  0.09049797  0.6888972   0.12416868]
  [ 0.6640018   0.10017505  0.6869478   0.11602707]
  [ 0.9438443   0.07688081  0.9538213   0.13743323]  

From my interpretation it should be the xmin, ymin, xmax and ymax (normalized by the image width and height of course). For the second element in the list : 0.99999875 0.9999977  0.99999756

My guess would be related to the score. For the third element I think it is related to the classes: 9.  9.  9.
With these assumptions I made this function to return the coordinates of the BB to draw them in the original image. The problem is mainly the output of this image: 

 ```
def convert_BB(output_data):

    boxes = output_data[0].numpy()[0]
    scores = output_data[1].numpy()[0]
    classes = output_data[2].numpy()[0]
    bbs=[]

    for i in range(0,100):
        bb=boxes[i]
        x_min, y_min, x_max, y_max = bb[0], bb[1], bb[2], bb[3]
        xyxy = [x_min, y_min, x_max, y_max]
        bbs.append(xyxy)

    return bbs, classes, scores
 ```
 `xyxy, classes, scores = convert_BB(predictions) `

 ```
from PIL import Image, ImageDraw

pil_image = Image.open("/content/dog_640.jpg")

# Create a drawing object
draw = ImageDraw.Draw(pil_image)

# Loop through the bounding boxes and draw rectangles
for i in range(len(scores)):
    if 0.5 < scores[i] <= 1.0:
        H, W = 640, 640
        xmin, ymin, xmax, ymax = (
            int(xyxy[i][0] * W),
            int(xyxy[i][1] * H),
            int(xyxy[i][2] * W),
            int(xyxy[i][3] * H),
        )

        # Draw a rectangle on the image
        draw.rectangle([(xmin, ymin), (xmax, ymax)], outline=(0, 255, 0), width=1) 

# Display or save the modified image
pil_image.save('/content/output_image.jpg')
 ```

So the outcome of this is nothing related to the preditcion of this image in the yolov5 repo:
 ![output_image (1)](https://github.com/ultralytics/yolov5/assets/70081290/fa70fd06-42a5-4f39-a186-d1c3ae37326d) 

Would really apreciate some help. When trying this method with a custom model the result is the the same, expect some of the values for the bounding box, and score are 0. And the classes -1. For now I would like to comprehend why these annoations are wrong. I think NMS probably as somthing to do with but not sure. Thank you so much.

**Comment by phsilvafct** [comment]
Hope too help someone in the future, really simple mistake: Just had to normalize the input image.

`normalized_image_array = input_image.astype(np.float32) / 255.0`

This fixed everything.

**Comment by glenn-jocher** [comment]
@phsilvafct glad to hear that you were able to solve the issue. Thank you for sharing the solution! It's great to see the community coming together to help each other. If you have any more questions or need further assistance, feel free to ask.
